### PR TITLE
Add Audio Effects demo

### LINKED ISSUE #673: Add an AudioEffect demo
As I mentioned in https://github.com/godotengine/godot-docs/pull/5470, we should probably create a demo where various audio effects can be played around with.

The demo should allow you to enable one or more effects. Background music should be toggleable (and played on a dedicated Music bus), and a sound should be playable by pressing a button (played on the Master bus).

### PR DESCRIPTION
**Pull request by dkleitsas**
Resolves #673 

A demo that showcases various Audio Effects available in Godot

### CODE REVIEW

File: audio/audio_effects_demo/audio_effects.gd [hunk 1]
```diff
@@ -0,0 +1,172 @@
+extends Node2D
+
+
+func _on_toggle_music_toggled(button_pressed):
+	if (button_pressed):
```
Comment (aaronfranke) [inline audio/audio_effects_demo/audio_effects.gd:5]:
```suggestion
	if button_pressed:
```

File: audio/audio_effects_demo/audio_effects.gd [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+		get_node("SoundEffects/AudioStreamPlayerMusic").play()
+	else:
+		get_node("SoundEffects/AudioStreamPlayerMusic").stop()
+
+
+
+func _on_ding_button_pressed():
```
Comment (aaronfranke) [inline audio/audio_effects_demo/audio_effects.gd:12]:
```suggestion
		get_node("SoundEffects/AudioStreamPlayerMusic").stop()


func _on_ding_button_pressed():
```

File: audio/audio_effects_demo/audio_effects.gd [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	get_node("SoundEffects/AudioStreamPlayerStatic").play()
+
+
+func _on_whistle_button_pressed():
+	get_node("SoundEffects/AudioStreamPlayerWhistle").play()
+
+
+
+func _on_toggle_amplify_toggled(button_pressed):
+	if(button_pressed):
```
Comment (aaronfranke) [inline audio/audio_effects_demo/audio_effects.gd:42]:
```suggestion
	if button_pressed:
```

There are more like this.

File: audio/audio_effects_demo/audio_effects.gd [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+		AudioServer.set_bus_effect_enabled(0,2,true)
+	else:
+		AudioServer.set_bus_effect_enabled(0,2,false)
+
+
+func _on_toggle_chorus_toggled(button_pressed):
+	if(button_pressed):
+		AudioServer.set_bus_effect_enabled(0,3,true)
+	else:
+		AudioServer.set_bus_effect_enabled(0,3,false)
```
Comment (aaronfranke) [inline audio/audio_effects_demo/audio_effects.gd:66]:
```suggestion
		AudioServer.set_bus_effect_enabled(0, 3, true)
	else:
		AudioServer.set_bus_effect_enabled(0, 3, false)
```

File: audio/audio_effects_demo/audio_effects.gd [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+func _on_toggle_stereo_enhance_toggled(button_pressed):
+	if(button_pressed):
+		AudioServer.set_bus_effect_enabled(0,17,true)
+	else:
+		AudioServer.set_bus_effect_enabled(0,17,false)
+
+
+
+
+
```
Comment (aaronfranke) [inline audio/audio_effects_demo/audio_effects.gd:172]:
```suggestion
```

The file should end in exactly one `\n` character.

**Comment by Calinou** [comment]
The folder's name should be renamed to `audio_effects`, as `demo` is redundant.

**Comment by dkleitsas** [comment]
I considered having sliders so you can adjust the parameters while it's running but I thought it would get a little cluttered since there's a lot of effects. Perhaps each one could have a little drop-down menu and the sliders be there. I'll try to figure that out when I have a little more time. Also, you are totally right I completely forgot the licensing, I had everything written down in a separate file and never included it, I will fix that asap.

**Comment by Calinou** [comment]
One concern I had was that the sounds aren't freely licensed, but we've tried to keep all the assets in https://github.com/godotengine/godot-demo-projects/ freely licensed so far.

Could you look for freely licensed sounds on [freesound.org](freesound.org/) to use as replacements? They must be under the CC0, CC BY or CC BY-SA licenses (*not* CC BY-NC-SA).

Sorry to ask for this so late, but it's something we should be careful about in the future (it's worth pointing out in `CONTRIBUTING.md`).

**Comment by dkleitsas** [comment]
Absolutely, I'll see what I can find.

**Comment by Calinou** [comment]
> Absolutely, I'll see what I can find.

Bump :slightly_smiling_face:

**Pull request closed without merging**